### PR TITLE
script/mafia is now properly atomic, this time for sure

### PR DESCRIPTION
**Pull request by jacobstanley**
This should in theory fix https://github.com/ambiata/mafia/issues/36

/cc @nhibberd

### CODE REVIEW

**Comment by olorin** [comment]
LGTM :+1:

**Comment by nhibberd** [comment]
:fallen_leaf:

File: script/mafia [hunk 1]
```diff
@@ -41,7 +43,11 @@ exec_mafia () {
       MAFIA_TEMP=$(mktemp -d 2>/dev/null || mktemp -d -t 'exec_mafia')
       trap "rm -rf \"$MAFIA_TEMP\"" EXIT
 
-      echo "Building mafia-$MAFIA_VERSION in $MAFIA_TEMP"
+      # Create a temporary file in MAFIA_BIN so we can do an atomic copy/move dance.
+      MAFIA_PATH_TEMP=$(mktemp --tmpdir=$MAFIA_BIN $MAFIA_FILE-XXXXXX 2>/dev/null || TMPDIR=$MAFIA_BIN mktemp -t $MAFIA_FILE)
+      trap "rm -f \"$MAFIA_PATH_TEMP\"" EXIT
```
Comment (charleso) [inline script/mafia:48]: I _think_ that this trap will override the one above. Would that be a problem?

Comment (jacobstanley) [inline script/mafia:48]: yeah that would be bad, it should probably be a single trap to a function call, will fix it up

**Pull request merged**